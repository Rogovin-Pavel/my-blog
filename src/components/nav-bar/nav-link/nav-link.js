import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const NavLink = ({ link_name, link_route }) => {
  return (
    <li className="navbar__item">
      <Link className="navbar__link" to={link_route}>{link_name}</Link>
    </li>
  );
};

NavLink.propTypes = {
  link_name: PropTypes.string,
  link_route: PropTypes.string,
};

export {
  NavLink
};