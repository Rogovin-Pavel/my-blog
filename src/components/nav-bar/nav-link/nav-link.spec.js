/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { NavLink } from './nav-link';
describe('Counter', () => {
  const link_name = 'HOME';
  const link_route = '/';

  test('snapshot renders', () => {
    const component = renderer.create(
      <Router>
        <NavLink 
          link_name={link_name}
          link_route={link_route}
        />
      </Router>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});