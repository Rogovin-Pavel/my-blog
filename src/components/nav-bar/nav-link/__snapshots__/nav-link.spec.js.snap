// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snapshot renders 1`] = `
<li
  className="navbar__item"
>
  <a
    className="navbar__link"
    href="/"
    onClick={[Function]}
  >
    HOME
  </a>
</li>
`;
